stdlib/multiprocessing/sharedctypes.pyi

```diff
@@ -1,43 +1,122 @@
+_CT = TypeVar("_CT", bound=_CData)
 
+@overload
+def RawValue(typecode_or_type: Type[_CT], *args: Any) -> _CT: ...
+@overload
+def RawValue(typecode_or_type: Union[str, Type[_CData]], *args: Any) -> Any: ...
+@overload
+def RawArray(typecode_or_type: Type[_CT], size_or_initializer: Union[int, Sequence[Any]]) -> ctypes.Array[_CT]: ...
+@overload
+def RawArray(typecode_or_type: Union[str, Type[_CData]], size_or_initializer: Union[int, Sequence[Any]]) -> Any: ...
+@overload
+def Value(typecode_or_type: Type[_CT], *args: Any, lock: Literal[False], ctx: Optional[BaseContext] = ...) -> _CT: ...
+@overload
+def Value(
+    typecode_or_type: Type[_CT], *args: Any, lock: Union[Literal[True], _LockLike], ctx: Optional[BaseContext] = ...
+) -> SynchronizedBase[_CT]: ...
+@overload
+def Value(
+    typecode_or_type: Union[str, Type[_CData]],
+    *args: Any,
+    lock: Union[Literal[True], _LockLike],
+    ctx: Optional[BaseContext] = ...,
+) -> SynchronizedBase[Any]: ...
+@overload
+def Value(
+    typecode_or_type: Union[str, Type[_CData]], *args: Any, lock: Union[bool, _LockLike] = ..., ctx: Optional[BaseContext] = ...
+) -> Any: ...
+@overload
+def Array(
+    typecode_or_type: Type[_CT],
+    size_or_initializer: Union[int, Sequence[Any]],
+    *,
+    lock: Literal[False],
+    ctx: Optional[BaseContext] = ...,
+) -> _CT: ...
+@overload
+def Array(
+    typecode_or_type: Type[_CT],
+    size_or_initializer: Union[int, Sequence[Any]],
+    *,
+    lock: Union[Literal[True], _LockLike],
+    ctx: Optional[BaseContext] = ...,
+) -> SynchronizedArray[_CT]: ...
+@overload
+def Array(
+    typecode_or_type: Union[str, Type[_CData]],
+    size_or_initializer: Union[int, Sequence[Any]],
+    *,
+    lock: Union[Literal[True], _LockLike],
+    ctx: Optional[BaseContext] = ...,
+) -> SynchronizedArray[Any]: ...
+@overload
 def Array(
     typecode_or_type: Union[str, Type[_CData]],
     size_or_initializer: Union[int, Sequence[Any]],
     *,
     lock: Union[bool, _LockLike] = ...,
     ctx: Optional[BaseContext] = ...,
-) -> _Array: ...
-def Value(
-    typecode_or_type: Union[str, Type[_CData]], *args: Any, lock: Union[bool, _LockLike] = ..., ctx: Optional[BaseContext] = ...
-) -> _Value: ...
+) -> Any: ...
+def copy(obj: _CT) -> _CT: ...
+@overload
+def synchronized(obj: _SimpleCData[_T], lock: Optional[_LockLike] = ..., ctx: Optional[Any] = ...) -> Synchronized[_T]: ...
+@overload
+def synchronized(obj: ctypes.Array[c_char], lock: Optional[_LockLike] = ..., ctx: Optional[Any] = ...) -> SynchronizedString: ...
+@overload
+def synchronized(obj: ctypes.Array[_CT], lock: Optional[_LockLike] = ..., ctx: Optional[Any] = ...) -> SynchronizedArray[_CT]: ...
+@overload
+def synchronized(obj: _CT, lock: Optional[_LockLike] = ..., ctx: Optional[Any] = ...) -> SynchronizedBase[_CT]: ...
+
+class _AcquireFunc(Protocol):
+    def __call__(self, block: bool = ..., timeout: Optional[float] = ...) -> bool: ...
+
+class SynchronizedBase(ContextManager[bool], Generic[_CT]):
```

Comment:
We don't derive from `ContextManager` or other protocols, unless the implementation does. Please just copy `__enter__` and the cumbersome `__exit__` over.